api/v1/nodescan/service.go

```diff
@@ -238,16 +240,35 @@ func (s *serviceImpl) getRuntimeVulns(containerRuntime *v1.GetNodeVulnerabilitie
 	return nil, nil
 }
 
-func (s *serviceImpl) GetNodeVulnerabilities(_ context.Context, req *v1.GetNodeVulnerabilitiesRequest) (*v1.GetNodeVulnerabilitiesResponse, error) {
-	if stringutils.AtLeastOneEmpty(req.GetKernelVersion(), req.GetOsImage()) {
-		return nil, status.Error(codes.InvalidArgument, "both os image and kernel version are required")
+func (s *serviceImpl) GetNodeVulnerabilities(ctx context.Context, req *v1.GetNodeVulnerabilitiesRequest) (*v1.GetNodeVulnerabilitiesResponse, error) {
+	resp := &v1.GetNodeVulnerabilitiesResponse{
```

Comment:
nit: this isn't needed until after the legacy check. Let's move this below that check